kit/feature/http_proxy.go

```diff
@@ -0,0 +1,65 @@
+package feature
+
+import (
+	"context"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"go.uber.org/zap"
+)
+
+// HTTPProxy is an HTTP proxy that's guided by a feature flag. If the feature flag
+// presented to it is enabled, it will perform the proxying behavior. Otherwise
+// it will be a no-op.
+type HTTPProxy struct {
+	proxy   *httputil.ReverseProxy
+	logger  *zap.Logger
+	enabler ProxyEnabler
+}
+
+// NewHTTPProxy returns a new Proxy.
+func NewHTTPProxy(dest *url.URL, logger *zap.Logger, enabler ProxyEnabler) *HTTPProxy {
+	return &HTTPProxy{
+		proxy:   newReverseProxy(dest, enabler.Key()),
+		logger:  logger,
+		enabler: enabler,
+	}
+}
+
+// Do performs the proxying. It returns whether or not the request was proxied.
+func (p *HTTPProxy) Do(w http.ResponseWriter, r *http.Request) bool {
```

Comment:
Did you specifically name this Do to avoid implementing http.Handler by naming it ServeHTTP? Just curious because that's what I would have done but there's really no reason to and I'm wondering if I'm missing any other tradeoffs.

Reply:
Yeah, I wanted to be able to express in the contract that the operation might be a no-op. It might not perform the proxying behavior based on the conditions its testing for. To do that I needed a return value. Once I had that, I figured there was no reason to stick with the `ServeHTTP` nomenclature here. This is a loosely held opinion though.

Reply:
Oh duh yeah the return 👍